runtime/src/main/java/com/dylibso/chicory/runtime/HostImports.java

```diff
@@ -91,4 +107,42 @@ public FromHost[] getIndex() {
     public void setIndex(FromHost[] index) {
         this.index = index;
     }
+
+    public HostImports resolve() {
+        var finalFunctions = Arrays.stream(functions).collect(Collectors.toSet());
+        var finalGlobals = Arrays.stream(globals).collect(Collectors.toSet());
+        var finalMemories = Arrays.stream(memories).collect(Collectors.toSet());
+        var finalTables = Arrays.stream(tables).collect(Collectors.toSet());
+
+        if (delegates != null) {
+            for (var delegate : delegates) {
+                var resolved = delegate.resolve();
+
+                for (var f : resolved.getFunctions()) {
+                    var matching =
+                            finalFunctions.stream()
+                                    .filter(
+                                            fun ->
+                                                    fun.getModuleName().equals(f.getModuleName())
+                                                            && fun.getFieldName()
+                                                                    .equals(f.getFieldName()))
+                                    .findFirst();
+                    if (matching.isPresent()) {
+                        if (!matching.get().override()) {
```

Comment:
I think we should switch this so that the controlling `override()` flag is on the function being added rather than the original one. As it stands now, this would mean that every host function would have to explicitly allow being overridden, and this would further mean that the decision to override a specific function or not would be removed from the end developer, but I think the end developers should be able to override any imported function if they choose.

I'd also suggest that we throw if the override is not applied rather than silently using the previously present one.

Reply:
I went over this a bit too quickly before and missed some important bits - need to set aside a few minutes to trace through it in detail.

Reply:
🙏 please consider that this is just "sketched up"